NRLMMD-GEOIPS/geoips
# 118 convert yaml files for gridlines interface

# Reviewer Checklist
https://github.com/NRLMMD-GEOIPS/.github/blob/main/.github/review-template.md

# Related Issues
fixes NRLMMD-GEOIPS/geoips#118

# Testing Instructions
To test if your yaml files fit to the yaml schema applied to them, use this line in a python editor
from geoips.schema import validate
then use the validate function to ensure that each converted yaml file fits to the base schema applied.
(See output for an example)

# Summary

From issue NRLMMD-GEOIPS/geoips#118: 2023-04-04, convert yaml files for gridlines interface
* Create yaml files for gridlines following gridlines schema
created: /geoips/interface_modules/gridlines/default.yaml
created: /geoips/interface_modules/gridlines/tc_0p25degree.yaml
created: /geoips/interface_modules/gridlines/tc_pmw.yaml
created: /geoips/interface_modules/gridlines/tc_visir_3200km.yaml
created: /geoips/interface_modules/gridlines/tc_visir.yaml
created: /geoips/interface_modules/gridlines/tc_windspeed.yaml

# Output
>>> from geoips.schema import validate
>>> from glob import glob
>>> files = glob('geoips/interface_modules/gridlines/*.yaml')
>>> files
['geoips/interface_modules/gridlines/default.yaml', 'geoips/interface_modules/gridlines/tc_0p25degree.yaml', 'geoips/interface_modules/gridlines/tc_pmw.yaml', 'geoips/interface_modules/gridlines/tc_windspeed.yaml', 'geoips/interface_modules/gridlines/tc_visir.yaml', 'geoips/interface_modules/gridlines/tc_visir_3200km.yaml']
>>> for file in files:
...     validate(file)
... 
{'interface': 'gridlines', 'metadata': {'name': 'default', 'family': 'standard', 'short-description': 'The default gridlines configuration', 'long-description': 'Top and left gridline labels, latitude and longitude lines colored\nblack, auto spacing, 1px linewidth, and [4, 2] linestyle\n'}, 'spec': {'labels': {'top': True, 'bottom': False, 'left': True, 'right': False}, 'latitude-lines': {'color': 'black', 'linestyle': [4, 2], 'linewidth': 1, 'spacing': 'auto'}, 'longitude-lines': {'color': 'black', 'linestyle': [4, 2], 'linewidth': 1, 'spacing': 'auto'}}}
{'interface': 'gridlines', 'metadata': {'name': 'tc_0p25degree', 'family': 'standard', 'short-description': 'TC 0.25 degree gridlines configuration', 'long-description': 'All gridline labels enabled, latitude and longitude lines colored\n#00B4B4B4, 0.25 degree spacing, 1px linewidth, and [4, 2] linestyle\n'}, 'spec': {'labels': {'top': True, 'bottom': True, 'left': True, 'right': True}, 'latitude-lines': {'color': '#00B4B4B4', 'linestyle': [4, 2], 'linewidth': 1, 'spacing': 0.25}, 'longitude-lines': {'color': '#00B4B4B4', 'linestyle': [4, 2], 'linewidth': 1, 'spacing': 0.25}}}
{'interface': 'gridlines', 'metadata': {'name': 'tc_pmw', 'family': 'standard', 'short-description': 'TC Passive Microwave gridlines configuration', 'long-description': 'All gridline labels enabled, latitude and longitude lines colored\n#00B4B4B4, 2 degree spacing, 1px linewidth, and [4, 2] linestyle\n'}, 'spec': {'labels': {'top': True, 'bottom': True, 'left': True, 'right': True}, 'latitude-lines': {'color': '#00B4B4B4', 'linestyle': [4, 2], 'linewidth': 1, 'spacing': 2}, 'longitude-lines': {'color': '#00B4B4B4', 'linestyle': [4, 2], 'linewidth': 1, 'spacing': 2}}}
{'interface': 'gridlines', 'metadata': {'name': 'tc_windspeed', 'family': 'standard', 'short-description': 'TC Windspeed gridlines configuration', 'long-description': 'All gridline labels enabled, latitude and longitude lines colored\n#00B4B4B4, 2 degree spacing, 1px linewidth, and [4, 2] linestyle\n'}, 'spec': {'labels': {'top': True, 'bottom': True, 'left': True, 'right': True}, 'latitude-lines': {'color': '#00B4B4B4', 'linestyle': [4, 2], 'linewidth': 1, 'spacing': 2}, 'longitude-lines': {'color': '#00B4B4B4', 'linestyle': [4, 2], 'linewidth': 1, 'spacing': 2}}}
{'interface': 'gridlines', 'metadata': {'name': 'tc_visir', 'family': 'standard', 'short-description': 'TC Visible Infrared gridlines configuration', 'long-description': 'All gridline labels enabled, latitude and longitude lines colored\n#000000, 2 degree spacing, 1px linewidth, and [4, 2] linestyle\n'}, 'spec': {'labels': {'top': True, 'bottom': True, 'left': True, 'right': True}, 'latitude-lines': {'color': '#000000', 'linestyle': [4, 2], 'linewidth': 1, 'spacing': 2}, 'longitude-lines': {'color': '#000000', 'linestyle': [4, 2], 'linewidth': 1, 'spacing': 2}}}
{'interface': 'gridlines', 'metadata': {'name': 'tc_visir_3200km', 'family': 'standard', 'short-description': 'TC Visible Infrared 3200km gridlines configuration', 'long-description': 'All gridline labels enabled, latitude and longitude lines colored\n#000000, 5 degree spacing, 1px linewidth, and [4, 2] linestyle\n'}, 'spec': {'labels': {'top': True, 'bottom': True, 'left': True, 'right': True}, 'latitude-lines': {'color': '#000000', 'linestyle': [4, 2], 'linewidth': 1, 'spacing': 5}, 'longitude-lines': {'color': '#000000', 'linestyle': [4, 2], 'linewidth': 1, 'spacing': 5}}}
